concourse/scripts/test_gpdb.py

```
@@ -34,8 +36,10 @@ def copy_output():
             print(  "======================================================================\n" +
                     "DIFF FILE: " + diff_file+"\n" +
                     "----------------------------------------------------------------------")
-            with open(diff_file, 'r') as fin:
-                print fin.read()
+            if os.path.exists(diff_file):
+                with open(diff_file, 'r') as fin:
```

I assume this was added to mitigate the existence race? If so, ignore the exception instead of doing a precheck -- EAFP is not only a Python best practice but also generally [a good way to deal with TOCTOU bugs](https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use#Preventing_TOCTOU). Otherwise, the race isn't fixed.

---

Ah, yeah, this check is not necessary. It is a hold-out from when I forced the build to fail before running `icw`

---

I'll remove it.